### PR TITLE
Module Correctness & Fixes + Mimalloc Cleanup

### PR DESCRIPTION
**Pull request by northaxosky**
No behavior change beyond the fixes.

## Broken features (silent logic bugs)
- **Facegen** : `keywordIsChildPlayer` was stored without a deref (function- vs data-pointer), so the
  child/player facegen exclusion was silently dead on every runtime. Deref once.
- **WorkbenchSound**: handler always early-returned (`IsEnter()` is always true since `kEnter==0`), and
  `As<TESFurniture>()` on a placed ref returns null. Fixed the enter/exit gate; resolve furniture via the
  base form.
- **DuplicateAddonNodeIndex**: `std::format("{{}}", msg)` was a literal `{}` (escaped braces, no
  placeholder), dropping the colliding-node list from the warning. Now `"{{{}}}"`.

## Thread / memory safety
- **ActorCauseSaveBloat**: `BSSpinLock{cell->spinLock}` copy-constructed a throwaway lock instead of
  acquiring the cell lock; `cell->references` iterated unsynchronized. Now `BSAutoLock`.
- **CombatMusic**: `ExecuteCommand` (main-thread-only) ran from a detached thread. Marshalled the burst
  to the main thread via `AddTask` + an atomic dedup flag.
- **MemoryManager**: `bhkThreadMemorySource` took `size_t&` where Havok's `hkMemoryAllocator` passes
  `std::int32_t&` (verified vs commonlibf4) → an 8-byte over-read of a 4-byte size on the buffer path.
  Matched the interface to `std::int32_t`.
- **WorkbenchSwap**: a bare `and dword[rdi+0x4]` on a `BSReadWriteLock::_lock` word (RE-confirmed
  contended, touched only by `lock cmpxchg`/`lock inc`) → non-atomic RMW could drop a concurrent update.
  Added a `lock` prefix.

## Cleanup
- Removed the vestigial `mimalloc` submodule + standalone vcxproj/filters

### CODE REVIEW

File: Addictol/Source/Modules/AdModuleFacegen.cpp [hunk 1]
```diff
@@ -237,7 +237,7 @@ namespace Addictol
 
 	bool FacegenSystem::InitContinue([[maybe_unused]] F4SE::MessagingInterface::Message* a_msg) noexcept
 	{
-		*(uintptr_t*)&keywordIsChildPlayer = REL::ID{ 533357, 2692125, 4799417 }.address();
+		keywordIsChildPlayer = REL::Relocation<RE::BGSKeyword*>{ REL::ID{ 533357, 2692125, 4799417 } }.get();
```
Comment (northaxosky) [inline Addictol/Source/Modules/AdModuleFacegen.cpp:240]: facegen stores data object pointer.

File: Addictol/Source/Modules/AdModuleWorkbenchSound.cpp [hunk 2]
```diff
@@ -59,11 +59,13 @@ namespace Addictol
 
 			RE::BSEventNotifyControl ProcessEvent(const RE::TESFurnitureEvent& a_event, RE::BSTEventSource<RE::TESFurnitureEvent>*)
 			{
-				if (a_event.IsEnter())
+				if (!a_event.IsExit())
```
Comment (northaxosky) [inline Addictol/Source/Modules/AdModuleWorkbenchSound.cpp:62]: `enum class FurnitureEventType : std::int32_t { kEnter = 0x0, kExit = 0x1 };`
`IsEnter()` is `type.all(kEnter)`. `TEnumSet::all(mask)` is `(bits & mask) == mask`, but `bits & 0 == 0` is always true

File: Addictol/Source/Modules/AdModuleWorkbenchSound.cpp [hunk 3]
```diff
@@ -59,11 +59,13 @@ namespace Addictol
...
 					// We only run on IsExit(), otherwise all workbenches would be silent
 					return RE::BSEventNotifyControl::kContinue;
 
-				RE::TESFurniture* furniture = a_event.targetFurniture.get()->As<RE::TESFurniture>();
+				RE::TESObjectREFR* furniterRef = a_event.targetFurniture.get();
```
Comment (northaxosky) [inline Addictol/Source/Modules/AdModuleWorkbenchSound.cpp]: `targetFurniture` is `NiPointer<TESObjectREFR>` which is the placed ref. `As<TESFurniture>()` always returns null because a ref cant be a furniture form. Need to go through the refs base object to reach the `TESFurniture`

File: Addictol/Source/Modules/AdModuleDuplicateAddonNodeIndex.cpp [hunk 4]
```diff
@@ -63,7 +63,7 @@ namespace Addictol
 
 				nodesErrorMessage += GetFormInfo(node);
 			}
-			nodesErrorMessage = std::format("{{}}"sv, nodesErrorMessage);
+			nodesErrorMessage = std::format("{{{}}}"sv, nodesErrorMessage);
```
Comment (northaxosky) [inline Addictol/Source/Modules/AdModuleDuplicateAddonNodeIndex.cpp:66]: in std::format {{ and }} are escaped literal braces

File: Addictol/Source/Modules/AdModuleActorCauseSaveBloat.cpp [hunk 5]
```diff
@@ -21,7 +21,7 @@ namespace Addictol
 			if (!a_cell)
 				return references_vector;
 
-			const auto lock = RE::BSSpinLock{ a_cell->spinLock };
+			const RE::BSAutoLock lock{ a_cell->spinLock };
```
Comment (northaxosky) [inline Addictol/Source/Modules/AdModuleActorCauseSaveBloat.cpp:24]: `BSSpinLock` is the mutex primitive, not a RAII guard. `BSAutoLock` is the RAII guard `BSAutoLock<BSSpinLock>`

File: Addictol/Source/Modules/AdModuleCombatMusic.cpp [hunk 6]
```diff
@@ -24,20 +24,23 @@ namespace Addictol
...
-
-				for (const std::string_view& command : commands)
-				{
-					ExecuteCommand(command, nullptr, true);
-				}
-			};
-
-			std::jthread t(asyncFunc);
-			t.detach();
+			if (fixQueued.exchange(true))
```
Comment (northaxosky) [inline Addictol/Source/Modules/AdModuleCombatMusic.cpp:31]: ExecuteCommand runs a console command through main thread only subsystems. Driving them from a detached worker is a data race against whatever the main thread might be doing. The `sleep_for` is fine but the `ExecuteCommand` burst is a violation. Mirrored how InteriorNavCut fix queues tasks.

File: Addictol/Source/Modules/AdModuleMemoryManager.cpp [hunk 7]
```diff
@@ -326,27 +326,27 @@ namespace Addictol
 			DeleteCriticalSection(&m_CritSec);
 		}
 
-		[[nodiscard]] virtual void* blockAlloc(size_t numBytes) noexcept
+		[[nodiscard]] virtual void* blockAlloc(std::int32_t numBytes) noexcept
```
Comment (northaxosky) [inline Addictol/Source/Modules/AdModuleMemoryManager.cpp:329]: matches commonlib's `hkMemoryAllocator`

File: Addictol/Source/Modules/AdModuleWorkbenchSwap.cpp [hunk 8]
```diff
@@ -15,6 +15,7 @@ namespace Addictol
 			{
 				Xbyak::Label retLab;
 
+				lock();
```
Comment (northaxosky) [inline Addictol/Source/Modules/AdModuleWorkbenchSwap.cpp:18]: [rdi+0x4] at 1573164+0x48 is a BSReadWriteLock::_lock

File: Addictol/Source/Modules/AdModuleFacegen.cpp (hunk 1 above)
Comment (Perchik71) [reply Addictol/Source/Modules/AdModuleFacegen.cpp:240]: in fact, it's the same, now it's in the libxse style.
